cmake/IWYU.cmake

```diff
@@ -0,0 +1,36 @@
+find_program(IWYU_PATH iwyu_tool)
+
+if(IWYU_PATH)
+    # iwyu is available
+
+    # add a target to inspect number of times headers are included.
+    add_custom_target(iwyu-list-includes
+        COMMAND grep -hor "\"#include .*\""  ${PROJECT_SOURCE_DIR}/src | sort | uniq -c | sort -n
+    )
+
+    # find the standard library directory
+    file(GLOB_RECURSE LLVM_INCLUDE /usr/lib*/llvm*/**/stddef.h)
+    if(LLVM_INCLUDE)
+        list(GET LLVM_INCLUDE 0 LLVM_INCLUDE)
+        get_filename_component(LLVM_INCLUDE "${LLVM_INCLUDE}" DIRECTORY)
+        set(LLVM_INCLUDE "-I${LLVM_INCLUDE}")
+    endif()
+
+    # add a target to run iwyu on all files (within the compilation database)
+    add_custom_target(iwyu-all
+        COMMAND ${IWYU_PATH} -o clang -p . -- -Xiwyu --cxx17ns ${LLVM_INCLUDE}
+    )
```

Comment:
What happens if `LLVM_INCLUDE` is empty?

Reply:
On my machine, it complains about not finding `stddef.h`. I'm not really sure why that is, though.